drivers/platform/aducm3029/irq.c

```diff
@@ -285,6 +344,10 @@ int32_t irq_global_enable(struct irq_ctrl_desc *desc)
 		irq_enable(desc, ADUCM_UART_INT_ID);
 	if (aducm_desc->enabled & (1u << ADUCM_RTC_INT_ID))
 		irq_enable(desc, ADUCM_RTC_INT_ID);
+	if (aducm_desc->enabled & (1u << ADUCM_GPIO_A_INT_ID))
+		irq_enable(desc, ADUCM_GPIO_A_INT_ID);
```

Comment:
You missed to implement irq_enable and irq_disable for gpio
I think you should use adi_gpio_GroupInterruptPolarityEnable

Reply:
Unfortunately I have not found a way to enable or disable the GPIO group interrupts and that is why I have left these functions not implemented. adi_gpio_GroupInterruptPolarityEnable seems to only changes the polarity of the interrupt trigger.

Reply:
My mistake again :)

Reply:
You can enable or disable the GPIO int with NVIC directly or by unregistering the callback

Reply:
Maybe with NVIC is faster, because otherwise you have to store the reference to the callback to re register it.

Reply:
I do prefer NVIC methods. I will try those.